### ISSUE TITLE
Real-time video frames utilization for skeleton decisions

### ISSUE THREAD

**Issue by yorgosk** [CMU-Perceptual-Computing-Lab/openpose#973]
Hello,

I am working on a ROS wrapper for the current version of OpenPose (future 1.5.0) and, because I want OpenPose to operate real-time, I try to reduce the latency as much as possible. I have a question about an OpenPose technical detail that I think I should take into account during my wrapper's operation.

How is OpenPose using the frames of a real-time video in order to decide the skeletons of the humans present to it? For instance, does it take single frames as fast as possible, which may mean that it drops certain frames while processing others, or does it take a "window" of them e.g. frames 0..29, 30..59, 60..89, etc / frames 0..29, 2..30, 3..3, etc. From what I have read so far and because of flags like ```--frame step``` I tend to believe that it does the former, taking single frames. However, I want to be sure that I fully understand this detail, because it may create opportunities for further speed improvements for my wrapper. As an extension to this question, what is the time efficiency with which OpenPose can handle a stream of images relatively connected with each other regarding their context vs a video?

If there is any published paper that addresses some of the above technical aspects I would also like to know.

Finally, because in older issue responses I have read that it is expected hand keypoint detection to noticeably slow down a real-time pose estimation process, should I still expect this to be the case? At least, this is what I observe while running OpenPose. 

Thanks in advance!

_Originally posted by @yorgosk in https://github.com/CMU-Perceptual-Computing-Lab/openpose/issues/754#issuecomment-447807056_

**Comment by gineshidalgo99** [comment]
I removed your other post to avoid duplicates. (Let me know if I skip any question)

1. OpenPose takes individual frames, it does not take any batch nor window.
2. I did not try any batch of images so far. For video, we will release a version of OpenPose that uses video information to speed up on videos/webcam (check the latest feature section periodically).
3. No paper about that that I am aware of.
4. Yes, hand and face are run after body and for each person, so they slow the speed considerably.
